Add support for external slurmdbd in Slurm configuration

jdeamicis opened the pull request:
### Description of changes
* Use ExternalSlurmdbd to render AccountingStorageHost when defined.

### Tests
* Add unit test case for Slurm config rendering to cover external slurmdbd scenario.
* Adapt other unit tests for Slurm configuration rendering when Slurm Accounting is used.
* Successful manual test to connect a cluster to a prototype external slurmdbd stack.

### References
* https://github.com/aws/aws-parallelcluster/pull/5926

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Add support for external slurmdbd in Slurm configuration

Use ExternalSlurmdbd to render AccountingStorageHost when defined.

Add unit test case to cover external slurmdbd scenario.

Adapt other unit tests for Slurm configuration rendering when Slurm
Accounting is used.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

jdeamicis committed:
Adapt Slurm config unit tests to external slurmdbd

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

NSsirena commented on test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf:
```diff
@@ -0,0 +1,18 @@
+AccountingStorageUser=slurm
+JobAcctGatherType=jobacct_gather/cgroup
+
+include <DIR>/pcluster/slurm_parallelcluster_efa_partition.conf
```
Why we need to include efa configuration in the slurm_dbd.conf ?

jdeamicis replied:
This is not the `slurmdbd.conf`, this is still the `slurm.conf`. But I see that the name of the file can be misleading (here I'm using `external_slurmdbd` as an appendix for the files in the external dbd case).
What appendix should we use here? `externaldbd`?

NSsirena replied:
Ahh.. ok. Yes, that is misleading.
Let's try with "_externaldbd"

jdeamicis replied:
Done. Resolving.

jdeamicis committed:
Change suffix for external dbd unit test case

The suffix `_external_slurmdbd` led to confusion with the
base name of the `slurmdbd.conf` include file generated
by ParallelCluster.

Switch to `_externaldbd` suffix to for more clarity.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>
